### ISSUE TITLE
FR: document implications of `--netfilter-mode=off` 

### ISSUE THREAD

**Issue by problame** [tailscale/tailscale#5283]
### What are you trying to do?

Use case: one of my hosts connects to iSCSI targets that have IPs in the `100.64.0.0/10` range.

A straight up `tailscale up` will install the iptables rule that drops all traffic from that range if it's not from tailscale, causing iSCSI timeouts and storage outage.

I tried `--netfilter=off` and it fixes the iSCSI timeout problem because the traffic isn't getting dropped (and the tailscale IP addresses are not conflicting with the iSCSI IP addresses, I guess).
Also, I can still connect to the services on that host, e.g., my monitoring host can still use tailscale to connect to the `node_exporter` running on the host that connects to the iSCSI targets.

The  wording in the docs suggest that it's unsafe to use `--netfilter=off`, though. So, I'm hesitant to let it running.

### How should we solve this?

Either provide more documentation on what the implications are of using `--netfilter=off`.

Or provide a way to make tailscale not mess with anything IPv4 related on a particular node (node should not even have a tailscale IPv4 address=.

Or provide a `--netfilter=no-drop-100.64.0.0` mode that doesn't install the DROP rule.

Or provide an additional flag to that adds exceptions to the drop rule, e.g. `--netfilter-allow=100.64.23.42` where 100.64.23.42 is my iSCSI hosts.

### What is the impact of not solving this?

I can't use tailscale on that host, except with `--netfilter=off`, whose implications are under-documented.

**Comment by DentonGentry** [comment]
> Or provide a way to make tailscale not mess with anything IPv4 related on a particular node (node should not even have a tailscale IPv4 address).

There is a `DisableIPv4` option in https://login.tailscale.com/admin/acls which may be useful.
https://tailscale.com/kb/1018/acls/#network-policy-options

**Comment by problame** [comment]
Sorry for taking so long to respond, I was on vacation.
Will `DisableIPv4` prevent routing of v4 subnets?

**Comment by DentonGentry** [comment]
> Will DisableIPv4 prevent routing of v4 subnets through tailscale?

Yes, it would.

**Comment by bsherman** [comment]
I'm also trying to find a workaround for the issue raised by @problame.
On the face of it, one would expect `DisableIPv4` to solve the problem. It does prevent the assigning of IPv4 address as stated in the docs, however, the docs also say:
> This option is intended for users who have a pre-existing conflicting use of the 100.64.0.0/10 carrier-grade NAT address range.

Simply setting this network policy option does not address the intent. White it does not assign IPv4 addresses, all IPv4 netfilter rules are still applied on the host (eg `iptables -A ts-input -s 100.64.0.0/10 ! -i tailscale0 -j DROP`) so anyone using CGNAT is still stuck.

Further, as mentioned above, one can use `tailscale up --netfilter=off` though, it seems wiser to use `--netfilter=nodivert` and then manually link only the desired ipv6 chains like so:
```
ip6tables -t nat -A POSTROUTING -j ts-postrouting
ip6tables -A INPUT -j ts-input
ip6tables -A FORWARD -j ts-forward
```

However, when doing this I see different behavior than when using `--netfilter=on`.  Specifically,  with `--netfilter=on` sending ipv6 traffic across tailscale0 on an advertised route results in masqueraded packets but I don't get masquerading when manually linking chains as show above.

In summary, it would be nice if the `DisableIPv4` network policy option just didn't touch the host's IPv4 network configuration at all.

**Comment by mrbluecoat** [comment]
Ref #11717

**Comment by clarkzjw** [comment]
So the `--netfilter-mode` option is only available on Linux https://tailscale.com/kb/1241/tailscale-up

I'm wondering what the equivalent workaround on macOS is to access ISP `100.64.0.0/10` addresses? For example, Starlink's gateway IP is `100.64.0.1` for IPv4. When Tailscale is on with `--netfilter-mode=on` on Linux or with the app open on macOS, users cannot ping `100.64.0.1`. It does not affect reaching the Internet over Starlink, however it's useful for troubleshooting or network measurement.  

Is it possible to workaround this with `pfctl` on macOS?

**Comment by Doublefire-Chen** [comment]
> So the `--netfilter-mode` option is only available on Linux https://tailscale.com/kb/1241/tailscale-up
> 
> I'm wondering what the equivalent workaround on macOS is to access ISP `100.64.0.0/10` addresses?

Same issue. I also want to find a way to make my macOS have access to `100.64.0.0/10` with tailscale.

**Comment by Sairyss** [comment]
By default when tailscale is up (with netfilter mode on) my internet becomes very slow, loading pages takes 5-10 seconds. When setting it to off with `--netfilter-mode=off` it fixes the issue.
It would be nice to have it documented to know what downsides are there when it's turned off and how to mitigate them.
